src/kirby/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -26,6 +28,17 @@ export class ModalWrapperComponent {
       providers: [{ provide: COMPONENT_PROPS, useValue: this.config.componentProps }],
       parent: injector,
     });
+
+    modalController.scrollToTop = this.scrollToTop.bind(this);
```

Comment:
Doesn't this prevent garbage collection?

Reply:
No idea, how would you test that?
We should probably update the `modalController.scrollToTop` when closing modals to avoid leaks as well as support scrolling still working after closing nested modal (we need to make the `scrollToTop` scroll on the right modal).

Reply:
This should be fixed now